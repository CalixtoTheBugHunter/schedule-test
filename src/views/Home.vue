<template>
  <div class="home">
    <base-button label="Save" />
  </div>
</template>

<script>
// @ is an alias to /src
import { BaseButton } from '@/components/atoms'

export default {
  name: 'Home',
  components: {
    BaseButton
  },
}
</script>
