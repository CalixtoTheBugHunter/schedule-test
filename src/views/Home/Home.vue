<template>
  <div class="Home">
    <section class="Home__schedule">
      <base-button label="New reminder" />
      <the-schedule />
    </section>
    <section class="Home__settings">
      <home-reminder />
    </section>
  </div>
</template>

<script>
import { BaseButton } from '@/components/atoms'
import { TheSchedule } from '@/components/organisms'
import HomeReminder from './HomeReminder/HomeReminder.vue'


export default {
  name: 'Home',
  components: { BaseButton, TheSchedule, HomeReminder }
}
</script>

<style lang="scss" scoped>
.Home {
  width: 100vw;
  height: 100vh;
  display: grid;
  grid-template-columns: 1fr 300px;

  &__settings {
    background: $color-light;
  }
}
</style>