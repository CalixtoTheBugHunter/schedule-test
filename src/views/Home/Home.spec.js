/* eslint-disable no-undef */
import { mount } from '@vue/test-utils'
import Home from './Home'
import { createStore } from 'vuex'

describe('Home', () => {
  let wrapper, store

  beforeEach(() => {

    store = createStore({
      modules: {
        schedule: {
          namespaced: true,
          state() {
            return {
              data: {}
            }
          },
          getters: {
            sheduleLength: () => 0
          }
        }
      }
    })

    wrapper = mount(Home, {
      global: {
        plugins: [store]
      }
    })
  })

  it('Component exists', () => {
    expect(wrapper.exists()).toBe(true)
  })

  it('Component not changed', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})