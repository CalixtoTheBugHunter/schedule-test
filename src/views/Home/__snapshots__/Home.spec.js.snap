// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component not changed 1`] = `
<div class="Home">
  <section class="Home__schedule">
    <base-button-stub label="New reminder"></base-button-stub>
    <the-schedule-stub></the-schedule-stub>
  </section>
  <section class="Home__settings">
    <home-reminder-stub></home-reminder-stub>
  </section>
</div>
`;
